### PR TITLE
Add plugin_info and healthz endpoints to s3 plugin service, add `chia data plugins check` cli

### PR DESCRIPTION
**Pull request by emlowe**
Add two new rest endpoints to the Datalayer S3 plugin service.

/healthz: {"success": true}
/plugin_info: {"name": name , "version": version_string, "instance": instance}

Also added `chia data plugins check` which iterates through the configured uploaders and downloaders and calls `/plugin_info`

Sample output:
```
❯ chia data plugins check
{
    "plugin_status": {
        "downloaders": {
            "http://localhost:8999": {
                "status": 404
            },
            "http://localhost:9847": {
                "Error": "ClientConnectorError Cannot connect to host localhost:9847 ssl:default [Connect call failed ('127.0.0.1', 9847)]"
            }
        },
        "uploaders": {
            "http://localhost:9002": {
                "response": {
                    "instance": "fullaccess",
                    "name": "Chia S3 Data Layer plugin",
                    "version": "0.1.0"
                },
                "status": 200
            }
        }
    },
    "success": true
}
```

### CODE REVIEW

File: chia/data_layer/s3_plugin_service.py [hunk 1]
```diff
@@ -87,6 +89,12 @@ async def upload(self, request: web.Request) -> web.Response:
             return web.json_response({"handles_url": False})
         return web.json_response({"uploaded": True})
 
+    async def plugin_id(self, request: web.Request) -> web.Response:
+        return web.json_response({"id": plugin_id, "version": plugin_version})
```
Comment (altendky) [inline chia/data_layer/s3_plugin_service.py:93]: Should `"success": True` be included everywhere that we respond successfully?

**Review by altendky — CHANGES_REQUESTED** [review]

File: chia/data_layer/s3_plugin_service.py [hunk 2]
```diff
@@ -17,6 +17,8 @@
 from chia.types.blockchain_format.sized_bytes import bytes32
 
 log = logging.getLogger(__name__)
+plugin_id = "2bdf9ed6-f3bc-44f6-a69d-06602c0688d9"
+plugin_version = "Chia S3 Datalayer plugin 0.1.0"
```
[the remark this answers is not in the document]
Comment (emlowe) [reply chia/data_layer/s3_plugin_service.py]: Changed a bit...

File: chia/data_layer/s3_plugin_service.py [hunk 3]
```diff
@@ -150,6 +158,8 @@ def make_app(config: Dict[str, Any], instance_name: str):  # type: ignore
     app.add_routes([web.post("/upload", s3_client.upload)])
     app.add_routes([web.post("/check_url", s3_client.check_url)])
     app.add_routes([web.post("/download", s3_client.download)])
+    app.add_routes([web.post("/plugin_id", s3_client.plugin_id)])
```
[the remark this answers is not in the document]
Comment (emlowe) [reply chia/data_layer/s3_plugin_service.py:205]: didn't find anything

File: chia/data_layer/s3_plugin_service.py (hunk 1 above)
Comment (emlowe) [reply chia/data_layer/s3_plugin_service.py]: no

**Review by altendky — DISMISSED** [review]

File: chia/data_layer/s3_plugin_service.py [hunk 4]
```diff
@@ -18,6 +18,9 @@
 from chia.types.blockchain_format.sized_bytes import bytes32
 
 log = logging.getLogger(__name__)
+plugin_id = "2bdf9ed6-f3bc-44f6-a69d-06602c0688d9"
```
Comment (almogdepaz) [inline chia/data_layer/s3_plugin_service.py:21]: this seems meaningless unless its dynamic since i assume each plugin will have its own version.

Comment (altendky) [reply chia/data_layer/s3_plugin_service.py:21]: Why would separate plugins be versioned to be unique from each other?  Though I suspect I'm just totally missing your intent.

File: chia/data_layer/s3_plugin_service.py [hunk 5]
```diff
@@ -122,6 +125,20 @@ async def upload(self, request: web.Request) -> web.Response:
             return web.json_response({"uploaded": False})
         return web.json_response({"uploaded": True})
 
+    async def plugin_id(self, request: web.Request) -> web.Response:
+        return web.json_response(
+            {
+                "id": plugin_id,
```
Comment (altendky) [inline chia/data_layer/s3_plugin_service.py:131]: ?

```suggestion
                "plugin_id": plugin_id,
```

File: chia/rpc/data_layer_rpc_api.py [hunk 6]
```diff
@@ -428,3 +429,10 @@ async def get_sync_status(self, request: Dict[str, Any]) -> EndpointResult:
                 "target_generation": sync_status.target_generation,
             }
         }
+
+    async def check_plugins(self, request: Dict[str, Any]) -> EndpointResult:
+        if self.service is None:
+            raise Exception("Data layer not created")
+        plugin_status = await self.service.check_plugins()
+
+        return {"plugin_status": {"uploaders": plugin_status.uploaders, "downloaders": plugin_status.downloaders}}
```
Comment (altendky) [inline chia/rpc/data_layer_rpc_api.py:438]: Should `PluginStatus` provide some marshalling for this?

File: chia/data_layer/s3_plugin_service.py (hunk 5 above)
Comment (emlowe) [reply chia/data_layer/s3_plugin_service.py:131]: Hmm this looks wrong, I changed this to /`plugin_info`

File: chia/data_layer/s3_plugin_service.py (hunk 4 above)
Comment (emlowe) [reply chia/data_layer/s3_plugin_service.py]: I removed this

File: chia/data_layer/data_layer.py [hunk 7]
```diff
@@ -863,3 +865,26 @@ async def get_uploaders(self, tree_id: bytes32) -> List[str]:
...
+        async def get_plugin_info(url: str) -> Dict[str, Any]:
+            async with aiohttp.ClientSession() as session:
+                try:
+                    async with session.post(url + "/plugin_info", json={}) as response:
+                        ret = {"status": response.status}
+                        if response.status == 200:
+                            ret["response"] = json.loads(await response.text())
+                        return ret
+                except Exception as e:
+                    return {"error": f"{type(e).__name__} {e}"}
```
[the remark this answers is not in the document]
Comment (emlowe) [reply chia/data_layer/data_layer.py]: Changed

File: chia/rpc/data_layer_rpc_api.py (hunk 6 above)
Comment (emlowe) [reply chia/rpc/data_layer_rpc_api.py]: added

File: chia/data_layer/data_layer.py [hunk 8]
```diff
@@ -863,3 +865,26 @@ async def get_uploaders(self, tree_id: bytes32) -> List[str]:
...
+
+        for uploader in self.uploaders:
+            uploader_status[uploader] = await get_plugin_info(uploader)
+
+        for downloader in self.downloaders:
+            downloader_status[downloader] = await get_plugin_info(downloader)
```
[the remark this answers is not in the document]
Comment (emlowe) [reply chia/data_layer/data_layer.py:889]: I'm inclined to just keep this simple at the moment.

File: chia/data_layer/data_layer.py [hunk 9]
```diff
@@ -863,3 +877,18 @@ async def get_uploaders(self, tree_id: bytes32) -> List[str]:
...
+        downloader_status = {}
+
+        coros = {asyncio.create_task(get_plugin_info(url=plugin)) for plugin in {*self.uploaders, *self.downloaders}}
+        results = await asyncio.gather(*coros)
+        for result in results:
+            url, status = result
+            if url in self.uploaders:
+                uploader_status[url] = status
+            if url in self.downloaders:
+                downloader_status[url] = status
```
Comment (altendky) [inline chia/data_layer/data_layer.py:892]:
```suggestion
        coros = [get_plugin_info(url=plugin)) for plugin in {*self.uploaders, *self.downloaders}]
        results = await asyncio.gather(*coros)
        
        uploader_status = {url: results.get(plugin, "unknown") for url in self.uploaders}
        downloader_status = {url: results.get(plugin, "unknown") for url in self.downloaders}
```

File: chia/data_layer/data_layer_util.py [hunk 10]
```diff
@@ -623,3 +623,9 @@ class SyncStatus:
     generation: int
     target_root_hash: bytes32
     target_generation: int
+
+
+@dataclasses.dataclass(frozen=True)
```
[the remark this answers is not in the document]
Comment (emlowe) [reply chia/data_layer/data_layer_util.py:628]: Acknowledge this comment about using frozen for a datalclass with mutable (eg Dict) members is a little curious

**Review by altendky — APPROVED** [review]

**Review by paninaro — APPROVED** [review]

**Pull request merged**